csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -73,61 +80,88 @@ public TArray Build(MemoryAllocator allocator = default)
             {
                 ValueOffsets.Append(Offset);
 
-                var data = new ArrayData(DataType, ValueOffsets.Length - 1, 0, 0,
-                    new[] { ArrowBuffer.Empty, ValueOffsets.Build(allocator), ValueBuffer.Build(allocator) });
+                var data = new ArrayData(DataType, ValueOffsets.Length - 1, NullCount, 0,
+                    new[] { ValidityBuffer.Build(allocator).ValueBuffer, ValueOffsets.Build(allocator), ValueBuffer.Build(allocator) });
 
                 return Build(data);
             }
 
-            public TBuilder Append(byte value)
+            public TBuilder AppendNull()
+            {
+                ValueOffsets.Append(Offset);
+                Offset++;
```

Comment:
Does `Offset` really need to be incremented here? We didn't append any values to `ValueBuffer`.

Reply:
We do not. You are correct.